### ISSUE TITLE
function in names repeated

### ISSUE THREAD

**Issue by Dieterbe** [grafana/metrictank#1676]
![function-names](https://user-images.githubusercontent.com/20774/74460391-7cda5b00-4e95-11ea-8d6a-b3d15d119dbe.png)

repeating this query `offset(scale(some.id.of.a.metric.{1,2,3,4},0),1)`
results in these names:
```

offset(scale(some.id.of.a.metric.1,0.000000),1.000000)
offset(scale(some.id.of.a.metric.2,0.000000),1.000000)
offset(scale(some.id.of.a.metric.3,0.000000),1.000000)
offset(scale(some.id.of.a.metric.4,0.000000),1.000000)
offset(scale(offset(scale(some.id.of.a.metric.1,0.000000),1.000000),0.000000),1.000000)
offset(scale(offset(scale(some.id.of.a.metric.2,0.000000),1.000000),0.000000),1.000000)
offset(scale(offset(scale(some.id.of.a.metric.3,0.000000),1.000000),0.000000),1.000000)
offset(scale(offset(scale(some.id.of.a.metric.4,0.000000),1.000000),0.000000),1.000000)


```

**Comment by shanson7** [comment]
I believe this was fixed in #1800

**Comment by Dieterbe** [comment]
Indeed, loosk good in current master (looks same as graphite)
![fixed](https://user-images.githubusercontent.com/20774/90010211-2a44f480-dca8-11ea-97d1-ba3174205bb2.png)
